data-prepper-plugins/kafka-connect-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafkaconnect/source/mongoDB/MongoDBSnapshotWorker.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafkaconnect.source.mongoDB;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import io.micrometer.core.instrument.Counter;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSet;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.opensearch.OpenSearchBulkActions;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.coordinator.SourceCoordinator;
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartition;
+import org.opensearch.dataprepper.plugins.kafkaconnect.configuration.MongoDBConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class MongoDBSnapshotWorker implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(MongoDBSnapshotWorker.class);
+    private static final Duration BACKOFF_ON_EXCEPTION = Duration.ofSeconds(60);
+    private static final Duration BACKOFF_ON_EMPTY_PARTITION = Duration.ofSeconds(60);
+    private static final Duration ACKNOWLEDGEMENT_SET_TIMEOUT = Duration.ofHours(2);
+    private static final String SUCCESS_ITEM_COUNTER_NAME = "exportRecordsSuccessTotal";
+    private static final String FAILURE_ITEM_COUNTER_NAME = "exportRecordsFailedTotal";
+    private static final String SUCCESS_PARTITION_COUNTER_NAME = "exportPartitionSuccessTotal";
+    private static final String FAILURE_PARTITION_COUNTER_NAME = "exportPartitionFailureTotal";
+    private static final String EVENT_SOURCE_COLLECTION_ATTRIBUTE = "__collection";
+    private static final String EVENT_SOURCE_DB_ATTRIBUTE = "__source_db";
```

Comment:
Any reason for this naming pattern ?

Reply:
As we agreed that we are running kafka connect for stream + initial snapshot together. 

This naming pattern is from Kafka Connect when streaming the data. We need to keep that consistant